src/engine/ruleset/decoders/sonicwall/firewall-generated.yml

```diff
@@ -0,0 +1,166 @@
+      - observer.product: Firewalls
+      - observer.type: Firewall
+      - observer.vendor: Sonicwall
+      - rsa.internal.messageid: $event.code
+      - rsa.time.event_time: $@timestamp
+      - rsa.network.dinterface: $observer.egress.interface.name
+      - rsa.network.sinterface: $observer.ingress.interface.name
+      - rsa.db.index: $~rsa.db.index
+      - tags: +a_append/forwarded/sonicwall.firewall
+      - rsa.identity.user_sid_dst: $~rsa.identity.user_sid_dst
+  - check:
+      - host.name: +ef_exists
+      - source.address: +ef_exists
+    map:
+      - related.hosts: +a_append/$host.name/$source.address
+  - check:
+      - destination.address: +ef_exists
+    map:
+      - related.hosts: +a_append/$destination.address
+      - rsa.network.host_dst: $destination.address
+  - check:
+      - source.ip: +ef_exists
+      - destination.ip: +ef_exists
+    map:
+      - related.ip: +a_append/$source.ip/$destination.ip
+  - check:
+      - source.nat.ip: +ef_exists
+      - destination.nat.ip: +ef_exists
+    map:
+      - related.ip: +a_append/$source.nat.ip/$destination.nat.ip
+  - check:
+      - ~log.start_message: +ef_not_exists
+    logpar:
+      - rsa.time.event_time: <~log.year>-<~log.mounth>-<~log.day>T<~log.fill>
+    map:
+      - aux_slash: '/'
+      - rsa.time.date: +s_concat/$~log.year/$aux_slash/$~log.mounth/$aux_slash/$~log.day
+      - aux_slash: +ef_delete
+  - check:
+      - ~log.note: +ef_exists
+    logpar:
+      - ~log.note: <~log.note.header> <~log.note.protocol.header>:<~log.note.protocol.payload>
+    map:
+      - network.protocol: $~log.note.protocol.payload
+  - check:
+      - ~log.fw_action: +ef_exists
+    map:
+      - event.action: $~log.fw_action
+      - rsa.internal.event_desc: $~log.msg_action
+  - check:
+      - ~log.fw_action: +ef_not_exists
+      - ~log.msg_action: accept
+    map:
+      - event.action: $~log.msg_action
+  - check:
+      - ~log.fw_action: +ef_not_exists
+      - ~log.msg_action: allow
+    map:
+      - event.action: $~log.msg_action
+  - check:
+      - ~log.fw_action: +ef_not_exists
+      - ~log.msg_action: block
+    map:
+      - event.action: $~log.msg_action
+  - check:
+      - ~log.fw_action: +ef_not_exists
+      - ~log.msg_action: cancel
+    map:
+      - event.action: $~log.msg_action
+  - check:
+      - ~log.fw_action: +ef_not_exists
+      - ~log.msg_action: deny
+    map:
+      - event.action: $~log.msg_action
+  - check: >-
+        +ef_not_exists/~log.fw_action AND ~log.msg_action!=accept AND
+        ~log.msg_action!=allow AND ~log.msg_action!=block AND
+        ~log.msg_action!=cancel AND ~log.msg_action!=deny
+    map:
+      - event.action: +ef_delete
+  - check:
+      - event.action: +ef_exists
+    map:
+      - rsa.misc.action: +a_append/$event.action
+  - check:
+      - event.action: +ef_not_exists
+    logpar:
+      - ~log.msg_action: "<~log.msg_action.payload>"
+    map:
+      - rsa.internal.msg: $~log.msg_action.payload
```

Comment:
Please, don't use `logpar` on the `normalize` stage. You could make use of the `s_trim` helper function in this case.

Reply:
I'll explain what I wanted to do here:
log.msg_action can return a single word such as "access" or it can return a phrase such as "access point". Only in the first case is it where the rsa.internal.msg field is mapped. In all other cases this field does not exist. I forgot to mention that another requirement is that event.action does not exist. Maybe it's a bit confusing but it's the logic I found in the expected json

Reply:
```
logpar:
      - ~log.msg_action: "<~log.msg_action.payload>"
```
This capture both cases as well, I don't think this is the correct approach to achieve it